components/social/org.wso2.carbon.social.core/src/main/java/org/wso2/carbon/social/core/ActivityPublisher.java

```diff
@@ -57,9 +57,23 @@ public long publish(String activity) throws SocialActivityException {
 		return publishActivity(jsonObject);
 	}
 
+	public String update(String activity) throws SocialActivityException {
+		JsonObject jsonActivity;
+		try {
+			jsonActivity = (JsonObject) parser.parse(activity);
+		} catch (JsonSyntaxException e) {
+			String message = "Malformed JSON element found: " + e.getMessage();
+			throw new SocialActivityException(message, e);
+		}
+		JsonObject updatedActivity = updateActivity(jsonActivity);
+		return updatedActivity.toString();
+	}
```

Comment:
Thanks @daneshk   , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/bf56e8d0d095fb59008cab25b9dfdf8963427541)